core/com.b2international.snowowl.api.rest/src/com/b2international/snowowl/api/rest/validation/ValidationRestService.java

```diff
@@ -0,0 +1,321 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.async.DeferredResult;
+
+import com.b2international.snowowl.api.rest.AbstractRestService;
+import com.b2international.snowowl.api.rest.admin.AbstractAdminRestService;
+import com.b2international.snowowl.api.rest.domain.RestApiError;
+import com.b2international.snowowl.api.rest.util.DeferredResults;
+import com.b2international.snowowl.api.rest.util.Responses;
+import com.b2international.snowowl.core.RepositoryInfo;
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.events.Request;
+import com.b2international.snowowl.core.events.util.Promise;
+import com.b2international.snowowl.core.exceptions.ApiValidation;
+import com.b2international.snowowl.core.exceptions.NotFoundException;
+import com.b2international.snowowl.core.internal.validation.ValidationConfiguration;
+import com.b2international.snowowl.core.validation.ValidateRequestBuilder;
+import com.b2international.snowowl.core.validation.ValidationRequests;
+import com.b2international.snowowl.core.validation.ValidationResult;
+import com.b2international.snowowl.core.validation.issue.ValidationIssue;
+import com.b2international.snowowl.core.validation.issue.ValidationIssues;
+import com.b2international.snowowl.core.validation.rule.ValidationRule;
+import com.b2international.snowowl.datastore.CodeSystemEntry;
+import com.b2international.snowowl.datastore.remotejobs.RemoteJobEntry;
+import com.b2international.snowowl.datastore.remotejobs.RemoteJobs;
+import com.b2international.snowowl.datastore.request.RepositoryRequests;
+import com.b2international.snowowl.datastore.request.job.JobRequests;
+import com.b2international.snowowl.terminologyregistry.core.request.CodeSystemRequests;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.annotations.ApiParam;
+import com.wordnik.swagger.annotations.ApiResponse;
+import com.wordnik.swagger.annotations.ApiResponses;
+
+/**
+ * Spring controller for exposing validation functionality.
+ * 
+ * @since 6.13
+ */
+@Api("Validations")
+@Controller
+@RequestMapping(produces={ AbstractRestService.SO_MEDIA_TYPE, MediaType.APPLICATION_JSON_VALUE })
+public class ValidationRestService extends AbstractAdminRestService {
+	
+	private Map<String, String> uniqueIdByUuid = Maps.newHashMap();
```

Comment:
Still not sure why do you need this map. 😕

Reply:
I guess a better alternative would be to compute an SHA-1 hash from the internal remove job ID and use that for the Validations API unique ID. You lose this state when you restart the server and that is bad.